diff --git a/application/modules/inventory/models/Index/Filter.php b/application/modules/inventory/models/Index/Filter.php
index fdad44d4f7..cb6101670f 100644
--- a/application/modules/inventory/models/Index/Filter.php
+++ b/application/modules/inventory/models/Index/Filter.php
@@ -142,7 +142,7 @@ class Inventory_Model_Index_Filter extends AbstractFilter
                 }
 
                 $rowData->description = strip_tags($rowData->description);
-                $rowData->project_title = App_Project_Repository::getInstance()->getProject($rowData->project_id)->getTitle();
+                $rowData->project_title = '123';
 
                 $rowData->is_active_formatted = $rowData->is_active == 1 ? 'не списан' : 'списан';
                 $rowData->type_title = App_Constant_Depot_Inventory_Type::getDescription()[(int) $rowData->type_id];
@@ -172,6 +172,7 @@ class Inventory_Model_Index_Filter extends AbstractFilter
             SELECT *
             FROM (
                 SELECT
+                    projects_depots.title,
                     0 AS is_free,
                     ext_inventory_accounting.id,
                     ext_inventory_depot_detailed.project_id,
@@ -282,10 +283,13 @@ class Inventory_Model_Index_Filter extends AbstractFilter
                 LEFT JOIN users u
                 ON 
                     ext_inventory_accounting.deactivation_user = u.id
+                LEFT JOIN projects_depots ON 
+                    projects_depots.id = ext_inventory_depot_detailed.depot_id 
     
                 UNION ALL
                 
                 SELECT 
+                    projects_depots.title,
                     1 AS is_free,
                     ext_inventory_depot_detailed.id AS id,
                     ext_inventory_depot_detailed.project_id,
@@ -374,6 +378,8 @@ class Inventory_Model_Index_Filter extends AbstractFilter
                     ext_inventory_depot_detailed.project_id = did2f_description.project_id AND
                     did2f_description.fieldId = 13 AND
                     did2f_description.id = ext_inventory_depot_2.field13
+                INNER JOIN projects_depots ON 
+                    projects_depots.id = ext_inventory_depot_detailed.depot_id 
                 WHERE ext_inventory_depot_detailed.free_amount > 0
             ) AS data
             {WHEREIN}
@@ -397,6 +403,8 @@ class Inventory_Model_Index_Filter extends AbstractFilter
             {WHERE}
         ";
 
+//        echo "<pre>" . print_r($sql, true); echo "</pre>"; // FrolovDEBUG
+//        exit();
         return $sql;
     }
 
diff --git a/library/App/Db/ExtInventoryDepotDetailed.php b/library/App/Db/ExtInventoryDepotDetailed.php
index 6423ad72d3..13d5c01c77 100644
--- a/library/App/Db/ExtInventoryDepotDetailed.php
+++ b/library/App/Db/ExtInventoryDepotDetailed.php
@@ -41,9 +41,9 @@ class App_Db_ExtInventoryDepotDetailed extends App_Db_Abstract
                     WHERE ext_inventory_depot_detailed_id = ext_inventory_depot_detailed.id
                 )
             '),
-            'free_amount' => new Zend_Db_Expr('amount-blocked')
+            'free_amount' => new Zend_Db_Expr('amount-blocked'),
         ], [
-            'id IN (?)' => $depotDetailed
+            'id IN (?)' => $depotDetailed,
         ]);
     }
 }
\ No newline at end of file
diff --git a/library/App/Filter/Component/Behaviour/Result.php b/library/App/Filter/Component/Behaviour/Result.php
index 91f1707b50..2d2e07d812 100644
--- a/library/App/Filter/Component/Behaviour/Result.php
+++ b/library/App/Filter/Component/Behaviour/Result.php
@@ -55,12 +55,18 @@ class App_Filter_Component_Behaviour_Result extends Behaviour
         }
 
         $sqlQueryForResult = $sqlFormatter->format('result', $filter->getResultSqlQuery());
+//        echo "<pre>" . print_r('$sqlQueryForResult', true); echo "</pre>"; // FrolovDEBUG
+//        echo "<pre>" . print_r($sqlQueryForResult, true); echo "</pre>"; // FrolovDEBUG
+//        exit();
         $sqlQueryForCount  = $sqlFormatter->format('count', $filter->getResultSqlQuery());
         
         $sqlQueryCount = $filter->getResultCountSqlQuery();
         $sqlQueryCount = str_replace('{RESULT}', $sqlQueryForCount, $sqlQueryCount);
 
         $sqlQuerySum = $filter->getResultSumSqlQuery();
+//        echo "<pre>" . print_r('$sqlQuerySum', true); echo "</pre>"; // FrolovDEBUG
+//        echo "<pre>" . print_r($sqlQuerySum, true); echo "</pre>"; // FrolovDEBUG
+//        exit();
 
         /** @var $db Zend_Db_Adapter_Abstract */
         if (!is_null($this->_filter->getDbAdapter())) {
@@ -73,6 +79,10 @@ class App_Filter_Component_Behaviour_Result extends Behaviour
 
         $result = $db->query($sqlQueryForResult)->fetchAll(Zend_Db::FETCH_OBJ);
 
+//        echo "<pre>" . print_r('$result', true); echo "</pre>"; // FrolovDEBUG
+//        echo "<pre>" . print_r($result, true); echo "</pre>"; // FrolovDEBUG
+//        exit();
+
         // весь результат
         if ($filter->useTotalResult) {
             $totalResult = $db->query($sqlQueryForCount)->fetchAll(Zend_Db::FETCH_OBJ);
diff --git a/library/App/Filter/Controller/AbstractFilterController.php b/library/App/Filter/Controller/AbstractFilterController.php
index d3ddf8723b..32cc1a6820 100644
--- a/library/App/Filter/Controller/AbstractFilterController.php
+++ b/library/App/Filter/Controller/AbstractFilterController.php
@@ -103,6 +103,8 @@ abstract class App_Filter_Controller_AbstractFilterController extends Zend_Contr
     public final function getfilterresultAction() {
         $filter = $this->_getFilter();
 
+//        echo "<pre>" . print_r($this, true); echo "</pre>"; // FrolovDEBUG
+//        exit();
         header('Content-Type: application/json');
         echo json_encode($filter->getResult());
         exit();
diff --git a/public/css/scss/app/application/sales-funnel/analytics.css b/public/css/scss/app/application/sales-funnel/analytics.css
index d7550fc1c9..b3a151fa79 100644
--- a/public/css/scss/app/application/sales-funnel/analytics.css
+++ b/public/css/scss/app/application/sales-funnel/analytics.css
@@ -15,7 +15,7 @@
   height: min-content;
 }
 .dashboard__row {
-  grid-column: 1/3;
+  grid-column: 1 / 3;
 }
 
 .dashboard-widget__title {
@@ -225,7 +225,7 @@
   justify-content: space-between;
 }
 .active-lead-lifetime-chart__control-cell_column-1 {
-  grid-row: 1/3;
+  grid-row: 1 / 3;
   grid-column: 1;
 }
 .active-lead-lifetime-chart__control-cell_1-2 {
diff --git a/public/js/app/application/inventory/service/create-fields.service.js b/public/js/app/application/inventory/service/create-fields.service.js
index de5cf66a4a..74fff0697d 100644
--- a/public/js/app/application/inventory/service/create-fields.service.js
+++ b/public/js/app/application/inventory/service/create-fields.service.js
@@ -11,6 +11,14 @@ define(["require", "exports"], function (require, exports) {
             align: 'center',
             alias: v => v == null ? '' : v
         },
+        title: {
+            position: 1,
+            caption: 'Склад',
+            sort: true,
+            align: 'center',
+            width: 100,
+            alias: v => parseInt(v) == 1 ? 'да' : 'нет'
+        },
         is_free: {
             position: 1,
             caption: 'Свободная запись',
diff --git a/typescript/app/application/inventory/service/create-fields.service.ts b/typescript/app/application/inventory/service/create-fields.service.ts
index 6af00595db..9a73f4bffc 100644
--- a/typescript/app/application/inventory/service/create-fields.service.ts
+++ b/typescript/app/application/inventory/service/create-fields.service.ts
@@ -14,6 +14,14 @@ export const DEFAULT_FIELDS: Assoc<TFilterFieldParamsOrdered<any>> = {
         align: 'center',
         alias: v => v == null ? '' : v
     },
+    title: {
+        position: 1,
+        caption: 'Склад',
+        sort: true,
+        align: 'center',
+        width: 100,
+        alias: v => parseInt(v) == 1 ? 'да' : 'нет'
+    },
     is_free: {
         position: 1,
         caption: 'Свободная запись',
