diff --git a/public/css/scss/app/application/sales-funnel/styles.css b/public/css/scss/app/application/sales-funnel/styles.css
index 653a5cfe0c..cc6febb9a6 100644
--- a/public/css/scss/app/application/sales-funnel/styles.css
+++ b/public/css/scss/app/application/sales-funnel/styles.css
@@ -2499,6 +2499,12 @@ command-assign-task-edit-params .tox .tox-editor-header {
   flex-direction: row;
   justify-content: space-between;
 }
+.amd-funnel-form__column-group {
+  display: flex;
+  gap: 16px;
+  border: 1px solid #d0d0d0;
+  border-radius: 4px;
+}
 .amd-funnel-form__group {
   padding: 5px;
   margin: 5px 10px 0 10px;
diff --git a/public/js/app/application/sales-funnel/funnel/component/funnel-form.component.js b/public/js/app/application/sales-funnel/funnel/component/funnel-form.component.js
index 1fdf37c215..c2c467f482 100644
--- a/public/js/app/application/sales-funnel/funnel/component/funnel-form.component.js
+++ b/public/js/app/application/sales-funnel/funnel/component/funnel-form.component.js
@@ -49,42 +49,44 @@ define(["require", "exports", "../../../../angular/decorator/component.decorator
     <amd-switch ng-model="$ctrl.model.generalConsiderVacations" ng-true-value="true" ng-false-value="false">
         <span class="amd-funnel-form__title amd-funnel-form__title_secondary">Учитывать сотрудников в отпуске</span>
     </amd-switch>
-    <div class="amd-funnel-form__row">
-        <div class="amd-funnel-form__column">
-            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к ролям</div>
-            <amd-zend-control data-errors="$ctrl.errors.filterRole">
+        <div class="amd-funnel-form__row">
+          <div class="amd-funnel-form__column-group">
+            <div class="amd-funnel-form__column">
+              <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к ролям</div>
+              <amd-zend-control data-errors="$ctrl.errors.filterRole">
                 <amd-select
-                    ng-if="$ctrl.filterRoleSource"
-                    data-value="$ctrl.model.filterRole"
-                    data-source="$ctrl.filterRoleSource"
-                    data-multiple="true"
-                ></amd-select>
-            </amd-zend-control>
-        </div>
-        <div class="amd-funnel-form__column">
-            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к отделам</div>
-            <amd-zend-control data-errors="$ctrl.errors.filterDepartment">
+                  ng-if="$ctrl.filterRoleSource"
+                  data-value="$ctrl.model.filterRole"
+                  data-source="$ctrl.filterRoleSource"
+                  data-multiple="true">
+                </amd-select>
+              </amd-zend-control>
+            </div>
+            <div class="amd-funnel-form__column">
+              <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к отделам</div>
+              <amd-zend-control data-errors="$ctrl.errors.filterDepartment">
                 <amd-select
-                    ng-if="$ctrl.filterDepartmentSource"
-                    data-value="$ctrl.model.filterDepartment"
-                    data-source="$ctrl.filterDepartmentSource"
-                    data-multiple="true"
-                ></amd-select>
-            </amd-zend-control>
-        </div>
-        <div class="amd-funnel-form__column">
+                  ng-if="$ctrl.filterDepartmentSource"
+                  data-value="$ctrl.model.filterDepartment"
+                  data-source="$ctrl.filterDepartmentSource"
+                  data-multiple="true">
+                </amd-select>
+              </amd-zend-control>
+            </div>
+          </div>
+          <div class="amd-funnel-form__column">
             <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Пользователи</div>
             <amd-zend-control data-errors="$ctrl.errors.filterUser">
-                <amd-select
-                    ng-if="$ctrl.filterUserSource"
-                    data-value="$ctrl.model.filterUser"
-                    data-source="$ctrl.filterUserSource"
-                    data-multiple="true"
-                    data-params="{selectAllFilteredByDefault: false}"
-                ></amd-select>
+              <amd-select
+                ng-if="$ctrl.filterUserSource"
+                data-value="$ctrl.model.filterUser"
+                data-source="$ctrl.filterUserSource"
+                data-multiple="true"
+                data-params="{selectAllFilteredByDefault: false}">
+              </amd-select>
             </amd-zend-control>
+          </div>
         </div>
-    </div>
 </div>
 
 <div class="amd-funnel-form__column amd-funnel-form__group margin-bottom_default">
@@ -122,6 +124,14 @@ define(["require", "exports", "../../../../angular/decorator/component.decorator
                 data-multiple="true"
             ></amd-select>
         </div>
+        <div class="amd-funnel-form__column">
+            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">По пользователям</div>
+            <amd-select
+                data-value="$ctrl.model.accessUsers"
+                data-source="$ctrl.userSource"
+                data-multiple="true"
+            ></amd-select>
+        </div>
     </div>
 </div>
 
diff --git a/scss/app/application/sales-funnel/_funnel-form.scss b/scss/app/application/sales-funnel/_funnel-form.scss
index cb4d6db88b..cb93769276 100644
--- a/scss/app/application/sales-funnel/_funnel-form.scss
+++ b/scss/app/application/sales-funnel/_funnel-form.scss
@@ -24,6 +24,12 @@
         justify-content: space-between;
     }
 
+    &__column-group {
+        display: flex;
+        gap: 16px;
+        border: 1px solid #d0d0d0;
+        border-radius: 4px;
+    }
     &__group {
         padding: 5px;
         margin: 5px 10px 0 10px;
diff --git a/typescript/app/application/sales-funnel/funnel/component/funnel-form.component.ts b/typescript/app/application/sales-funnel/funnel/component/funnel-form.component.ts
index 51a6a1f029..279cbb12c3 100644
--- a/typescript/app/application/sales-funnel/funnel/component/funnel-form.component.ts
+++ b/typescript/app/application/sales-funnel/funnel/component/funnel-form.component.ts
@@ -53,42 +53,44 @@ const TABLE = `
     <amd-switch ng-model="$ctrl.model.generalConsiderVacations" ng-true-value="true" ng-false-value="false">
         <span class="amd-funnel-form__title amd-funnel-form__title_secondary">Учитывать сотрудников в отпуске</span>
     </amd-switch>
-    <div class="amd-funnel-form__row">
-        <div class="amd-funnel-form__column">
-            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к ролям</div>
-            <amd-zend-control data-errors="$ctrl.errors.filterRole">
+        <div class="amd-funnel-form__row">
+          <div class="amd-funnel-form__column-group">
+            <div class="amd-funnel-form__column">
+              <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к ролям</div>
+              <amd-zend-control data-errors="$ctrl.errors.filterRole">
                 <amd-select
-                    ng-if="$ctrl.filterRoleSource"
-                    data-value="$ctrl.model.filterRole"
-                    data-source="$ctrl.filterRoleSource"
-                    data-multiple="true"
-                ></amd-select>
-            </amd-zend-control>
-        </div>
-        <div class="amd-funnel-form__column">
-            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к отделам</div>
-            <amd-zend-control data-errors="$ctrl.errors.filterDepartment">
+                  ng-if="$ctrl.filterRoleSource"
+                  data-value="$ctrl.model.filterRole"
+                  data-source="$ctrl.filterRoleSource"
+                  data-multiple="true">
+                </amd-select>
+              </amd-zend-control>
+            </div>
+            <div class="amd-funnel-form__column">
+              <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Принадлежность к отделам</div>
+              <amd-zend-control data-errors="$ctrl.errors.filterDepartment">
                 <amd-select
-                    ng-if="$ctrl.filterDepartmentSource"
-                    data-value="$ctrl.model.filterDepartment"
-                    data-source="$ctrl.filterDepartmentSource"
-                    data-multiple="true"
-                ></amd-select>
-            </amd-zend-control>
-        </div>
-        <div class="amd-funnel-form__column">
+                  ng-if="$ctrl.filterDepartmentSource"
+                  data-value="$ctrl.model.filterDepartment"
+                  data-source="$ctrl.filterDepartmentSource"
+                  data-multiple="true">
+                </amd-select>
+              </amd-zend-control>
+            </div>
+          </div>
+          <div class="amd-funnel-form__column">
             <div class="amd-funnel-form__title amd-funnel-form__title_secondary">Пользователи</div>
             <amd-zend-control data-errors="$ctrl.errors.filterUser">
-                <amd-select
-                    ng-if="$ctrl.filterUserSource"
-                    data-value="$ctrl.model.filterUser"
-                    data-source="$ctrl.filterUserSource"
-                    data-multiple="true"
-                    data-params="{selectAllFilteredByDefault: false}"
-                ></amd-select>
+              <amd-select
+                ng-if="$ctrl.filterUserSource"
+                data-value="$ctrl.model.filterUser"
+                data-source="$ctrl.filterUserSource"
+                data-multiple="true"
+                data-params="{selectAllFilteredByDefault: false}">
+              </amd-select>
             </amd-zend-control>
+          </div>
         </div>
-    </div>
 </div>
 
 <div class="amd-funnel-form__column amd-funnel-form__group margin-bottom_default">
@@ -126,6 +128,14 @@ const TABLE = `
                 data-multiple="true"
             ></amd-select>
         </div>
+        <div class="amd-funnel-form__column">
+            <div class="amd-funnel-form__title amd-funnel-form__title_secondary">По пользователям</div>
+            <amd-select
+                data-value="$ctrl.model.accessUsers"
+                data-source="$ctrl.userSource"
+                data-multiple="true"
+            ></amd-select>
+        </div>
     </div>
 </div>
 
