diff --git a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
index 425c26ab26..cea244200e 100644
--- a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
+++ b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
@@ -140,7 +140,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
 
         $subscriber = new LeadActivitySubscriber($this->leadFactoryMock, $this->leadCrudMock);
 
-        $this->eventService->setImmediateSubscriberClassList([]);
+        $this->eventService->setImmediateSubscriberClassList([$subscriber::class]);
         $this->eventService->getDispatcher()->addSubscriber($subscriber);
         authAsSystem();
 
@@ -286,132 +286,144 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         foreach ($events as $key => $alias) {
             $this->createdEvents[] = $alias;
             $eventGeneration = match ($alias) {
-                App\Event\SalesFunnel\Type\LeadCreation::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadCreation(
-                            new App\Event\SalesFunnel\Subject\Lead()
-                                ->setLeadId($lead->getEntity()->getId())
-                        ),
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\OwnershipTransfer\RequestCreated::class => function () use ($lead) {
-                    $request = $this->requestFactory->emptyModel();
-                    $request->getEntity()
-                        ->setLeadId($lead->getEntity()->getId())
-                        ->setNewOwnerId(1160)
-                        ->setStateId(App_Constant_Table_SalesFunnelLeadOwnershipTransferState::TYPE_ACTIVE)
-                        ->setCreatedBy(App_User_Wrapper::getInstance()->getId())
-                        ->setCreatedAt(new DateTimeImmutable());
-
-                    $requestEntity = $this->requestRepository->save($request->getEntity());
-
-                    $subject = new OwnershipTransferSubject()
-                        ->setRequestId($requestEntity->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\OwnershipTransfer\RequestCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadStageChanged::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadStageChanged(
-                            new App\Event\SalesFunnel\Subject\LeadField()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setField('stageId')
-                                ->setOldValue(459)
-                                ->setNewValue(460)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadClientChanged::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadClientChanged(
-                            new App\Event\SalesFunnel\Subject\LeadField()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setNewValue(null)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadFieldChanged::class => array_map(function ($field) use ($lead) {
-                    return function () use (
-                        $lead,
-                        $field
-                    ) {
-                        $this->eventGenerate($lead,
-                            new App\Event\SalesFunnel\Type\LeadFieldChanged(
-                                new App\Event\SalesFunnel\Subject\LeadField()
-                                    ->setLeadId($lead->getEntity()->getId())
-                                    ->setField($field)
-                                    ->setOldValue(null)
-                                    ->setNewValue(null)
-                            )
-                        );
-                    };
-                }, $fields),
-
-                App\Event\SalesFunnel\Type\LeadOwnerChanged::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadOwnerChanged(
-                            new App\Event\SalesFunnel\Subject\LeadField()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setField(null)
-                                ->setOldValue(null)
-                                ->setNewValue(null)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadAssigneeChanged::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadAssigneeChanged(
-                            new App\Event\SalesFunnel\Subject\LeadField()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setField(null)
-                                ->setOldValue(null)
-                                ->setNewValue(null)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\QuantityControl\LeadConfirmed::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\QuantityControl\LeadConfirmed(
-                            new App\Event\SalesFunnel\Subject\LeadField()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setField('isConfirmed')
-                                ->setOldValue(null)
-                                ->setNewValue(null)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadPhoneAdded::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadPhoneAdded(
-                            new App\Event\SalesFunnel\Subject\LeadPhone()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setPhone(null)
-                                ->setExt(null)
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\LeadPhoneRemoved::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\LeadPhoneRemoved(
-                            new App\Event\SalesFunnel\Subject\LeadPhone()
-                                ->setLeadId($lead->getEntity()->getId())
-                                ->setLeadPhoneId(null)
-                                ->setPhone(null)
-                                ->setExt(null)
-                        )
-                    );
-                },
-
+// ----- WORKED POSITIVE
+//                App\Event\SalesFunnel\Type\LeadCreation::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadCreation(
+//                            new App\Event\SalesFunnel\Subject\Lead()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                        ),
+//                    );
+//                },
+// ----- WORKED POSITIVE
+//                App\Event\SalesFunnel\Type\OwnershipTransfer\RequestCreated::class => function () use ($lead) {
+//                    $request = $this->requestFactory->emptyModel();
+//                    $request->getEntity()
+//                        ->setLeadId($lead->getEntity()->getId())
+//                        ->setNewOwnerId(1160)
+//                        ->setStateId(App_Constant_Table_SalesFunnelLeadOwnershipTransferState::TYPE_ACTIVE)
+//                        ->setCreatedBy(App_User_Wrapper::getInstance()->getId())
+//                        ->setCreatedAt(new DateTimeImmutable());
+//
+//                    $requestEntity = $this->requestRepository->save($request->getEntity());
+//
+//                    $subject = new OwnershipTransferSubject()
+//                        ->setRequestId($requestEntity->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\OwnershipTransfer\RequestCreated($subject)
+//                    );
+//                },
+//
+// ----- WORKED POSITIVE
+
+//                App\Event\SalesFunnel\Type\LeadStageChanged::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadStageChanged(
+//                            new App\Event\SalesFunnel\Subject\LeadField()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setField('stageId')
+//                                ->setOldValue(459)
+//                                ->setNewValue(460)
+//                        )
+//                    );
+//                },
+//
+// ----- WORKED POSITIVE
+
+//                App\Event\SalesFunnel\Type\LeadClientChanged::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadClientChanged(
+//                            new App\Event\SalesFunnel\Subject\LeadField()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setNewValue(null)
+//                        )
+//                    );
+//                },
+//
+// ----- WORKED POSITIVE
+
+//                App\Event\SalesFunnel\Type\LeadFieldChanged::class => array_map(function ($field) use ($lead) {
+//                    return function () use (
+//                        $lead,
+//                        $field
+//                    ) {
+//                        $this->eventGenerate($lead,
+//                            new App\Event\SalesFunnel\Type\LeadFieldChanged(
+//                                new App\Event\SalesFunnel\Subject\LeadField()
+//                                    ->setLeadId($lead->getEntity()->getId())
+//                                    ->setField($field)
+//                                    ->setOldValue(null)
+//                                    ->setNewValue(null)
+//                            )
+//                        );
+//                    };
+//                }, $fields),
+
+//            ---- UNWORKED POSITIVE
+//                App\Event\SalesFunnel\Type\LeadOwnerChanged::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadOwnerChanged(
+//                            new App\Event\SalesFunnel\Subject\LeadField()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setField(null)
+//                                ->setOldValue(null)
+//                                ->setNewValue(null)
+//                        )
+//                    );
+//                },
+
+
+
+//            WORKED POSITIVE ----
+//                App\Event\SalesFunnel\Type\LeadAssigneeChanged::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadAssigneeChanged(
+//                            new App\Event\SalesFunnel\Subject\LeadField()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setField(null)
+//                                ->setOldValue(null)
+//                                ->setNewValue(null)
+//                        )
+//                    );
+//                },
+
+//
+//                App\Event\SalesFunnel\Type\QuantityControl\LeadConfirmed::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\QuantityControl\LeadConfirmed(
+//                            new App\Event\SalesFunnel\Subject\LeadField()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setField('isConfirmed')
+//                                ->setOldValue(null)
+//                                ->setNewValue(null)
+//                        )
+//                    );
+//                },
+
+//                App\Event\SalesFunnel\Type\LeadPhoneAdded::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadPhoneAdded(
+//                            new App\Event\SalesFunnel\Subject\LeadPhone()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setPhone(null)
+//                                ->setExt(null)
+//                        )
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\LeadPhoneRemoved::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\LeadPhoneRemoved(
+//                            new App\Event\SalesFunnel\Subject\LeadPhone()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                                ->setLeadPhoneId(null)
+//                                ->setPhone(null)
+//                                ->setExt(null)
+//                        )
+//                    );
+//                },
+//
                 App\Event\SalesFunnel\Type\LeadEmailAdded::class => function () use ($lead) {
                     $this->eventGenerate($lead,
                         new App\Event\SalesFunnel\Type\LeadEmailAdded(
@@ -422,7 +434,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         )
                     );
                 },
-
+//
                 App\Event\SalesFunnel\Type\LeadEmailRemoved::class => function () use ($lead) {
                     $this->eventGenerate($lead,
                         new App\Event\SalesFunnel\Type\LeadEmailRemoved(
@@ -433,7 +445,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         )
                     );
                 },
-
+//
                 App\Event\SalesFunnel\Type\SurveyAnswerAdded::class => function () use ($lead) {
                     $this->eventGenerate($lead,
                         new App\Event\SalesFunnel\Type\SurveyAnswerAdded(
@@ -442,304 +454,307 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         )
                     );
                 },
-
-                App\Event\SalesFunnel\Type\RelationWithCallCreated::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\RelationWithCallCreated(
-                            new App\Event\SalesFunnel\Subject\LeadRelation()
-                                ->setLeadId($lead->getEntity()->getId())
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\MissedCallCreated::class => function () use ($lead) {
-                    $this->eventGenerate($lead,
-                        new App\Event\SalesFunnel\Type\Relation\MissedCallCreated(
-                            new App\Event\SalesFunnel\Subject\LeadRelation()
-                                ->setLeadId($lead->getEntity()->getId())
-                        )
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\InvoiceCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\InvoiceCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\OfferCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\OfferCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CalculationSheetCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CalculationSheetCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\AgreementListCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\AgreementListCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ContractCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ContractCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ClaimCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ClaimCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ClientAddRequestCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ClientAddRequestCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CashAccountCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CashAccountCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CashMoneyCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CashMoneyCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\AvitoChatMessageCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\EdnaMessageCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\EdnaMessageCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\RelationWithCallRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\RelationWithCallRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\InvoiceRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\InvoiceRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\OfferRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\OfferRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CalculationSheetRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CalculationSheetRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\AgreementListRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\AgreementListRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ContractRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ContractRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ClaimRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ClaimRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ClientAddRequestRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ClientAddRequestRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CashAccountRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CashAccountRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\CashMoneyRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\CashMoneyRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\AvitoChatMessageRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksRemoved($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\Relation\MissedCallDeleted::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\Relation\MissedCallDeleted($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\RelationWithMailServerMessageCreated::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\RelationWithMailServerMessageCreated($subject)
-                    );
-                },
-
-                App\Event\SalesFunnel\Type\RelationWithMailServerMessageRemoved::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\RelationWithMailServerMessageRemoved($subject)
-                    );
-                },
-
+//            WORKED POSITIVE ----
+
+//                App\Event\SalesFunnel\Type\RelationWithCallCreated::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\RelationWithCallCreated(
+//                            new App\Event\SalesFunnel\Subject\LeadRelation()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                        )
+//                    );
+//                },
+
+//                App\Event\SalesFunnel\Type\Relation\MissedCallCreated::class => function () use ($lead) {
+//                    $this->eventGenerate($lead,
+//                        new App\Event\SalesFunnel\Type\Relation\MissedCallCreated(
+//                            new App\Event\SalesFunnel\Subject\LeadRelation()
+//                                ->setLeadId($lead->getEntity()->getId())
+//                        )
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\InvoiceCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\InvoiceCreated($subject)
+//                    );
+//                },
+
+//                App\Event\SalesFunnel\Type\Relation\OfferCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\OfferCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CalculationSheetCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CalculationSheetCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\AgreementListCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\AgreementListCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ContractCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ContractCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ClaimCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ClaimCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ClientAddRequestCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ClientAddRequestCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CashAccountCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CashAccountCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CashMoneyCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CashMoneyCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\AvitoChatMessageCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\EdnaMessageCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\EdnaMessageCreated($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\RelationWithCallRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\RelationWithCallRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\InvoiceRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\InvoiceRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\OfferRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\OfferRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CalculationSheetRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CalculationSheetRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\AgreementListRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\AgreementListRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ContractRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ContractRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ClaimRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ClaimRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ClientAddRequestRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ClientAddRequestRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CashAccountRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CashAccountRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\CashMoneyRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\CashMoneyRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\AvitoChatMessageRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesRemoved($subject)
+//                    );
+//                },
+////
+//                App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksRemoved($subject)
+//                    );
+//                },
+//
+//                App\Event\SalesFunnel\Type\Relation\MissedCallDeleted::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\Relation\MissedCallDeleted($subject)
+//                    );
+//                },
+
+//                App\Event\SalesFunnel\Type\RelationWithMailServerMessageCreated::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\RelationWithMailServerMessageCreated($subject)
+//                    );
+//                },
+
+//                App\Event\SalesFunnel\Type\RelationWithMailServerMessageRemoved::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\RelationWithMailServerMessageRemoved($subject)
+//                    );
+//                },
+
+
+//            WORKED POSITIVE ---
                 App\Event\SalesFunnel\Type\ChildLeadCreated::class => function () use ($lead) {
                     $subject = new App\Event\SalesFunnel\Subject\ChildLead()
                         ->setLeadId($lead->getEntity()->getId())
@@ -805,7 +820,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         new App\Event\SalesFunnel\Type\Channel\FailedChannelMessageEvent($subject)
                     );
                 },
-
+//
                 App\Event\SalesFunnel\Type\LeadPropValueChanged::class => function () use ($lead) {
                     $subject = (new App\Event\SalesFunnel\Subject\LeadPropValue())
                         ->setLeadId($lead->getEntity()->getId());
@@ -815,7 +830,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         new App\Event\SalesFunnel\Type\LeadPropValueChanged($subject)
                     );
                 },
-
+//
                 App\Event\SalesFunnel\Type\LeadMergeStart::class => function () use ($lead) {
                     $subject = new App\Event\SalesFunnel\Subject\LeadMerge()
                         ->setTargetLeadId($lead->getEntity()->getId())
@@ -827,7 +842,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                         new App\Event\SalesFunnel\Type\LeadMergeStart($subject)
                     );
                 },
-
+//
                 App\Event\SalesFunnel\Type\LeadRemoved::class => function () use ($lead) {
                     $subject = new App\Event\SalesFunnel\Subject\LeadRemoved()
                         ->setLeadId($lead->getEntity()->getId());
@@ -838,14 +853,14 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                     );
                 },
 
-                App\Event\SalesFunnel\Type\LeadClosed::class => function () use ($lead) {
-                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
-                        ->setLeadId($lead->getEntity()->getId());
-                    $this->eventGenerate(
-                        $lead,
-                        new App\Event\SalesFunnel\Type\LeadClosed($subject)
-                    );
-                },
+//                App\Event\SalesFunnel\Type\LeadClosed::class => function () use ($lead) {
+//                    $subject = new App\Event\SalesFunnel\Subject\LeadRelation()
+//                        ->setLeadId($lead->getEntity()->getId());
+//                    $this->eventGenerate(
+//                        $lead,
+//                        new App\Event\SalesFunnel\Type\LeadClosed($subject)
+//                    );
+//                },
 
                 default => null,
             };
