diff --git a/library/App/Service/EventService.php b/library/App/Service/EventService.php
index dfe3cc1012..fbf8ccb7ba 100644
--- a/library/App/Service/EventService.php
+++ b/library/App/Service/EventService.php
@@ -395,4 +395,10 @@ class EventService
 
         return $normSubscriberSettings;
     }
+
+
+    public function setDefferedSubscriberClassList(array $classList)
+    {
+        $this->defferedSubscriberClassList = $classList;
+    }
 }
diff --git a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
index 75ab2adaaf..f07ce643b0 100644
--- a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
+++ b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
@@ -12,6 +12,8 @@ require_once "Mocks/RelationManagerMock.php";
 require_once "Mocks/LeadFactoryMock.php";
 require_once "Mocks/StageFactoryMock.php";
 
+use App\Event\SalesFunnel\Subject\LeadEmail;
+use App\Event\SalesFunnel\Subject\OwnershipTransfer as OwnershipTransferSubject;
 use App\SalesFunnel\Funnel\Repository as Repository;
 use App\SalesFunnel\Lead\Model;
 use App\SalesFunnel\Lead\Services\Crud;
@@ -32,9 +34,6 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
             App\SalesFunnel\EventSubscriber\LeadActivitySubscriber::class
         ];
 
-        echo "<pre>" . print_r($subscribers, true); echo "</pre>"; // FrolovDEBUG
-        exit();
-
         $this->eventService->setDefferedSubscriberClassList($subscribers);
 
         echo "Setup test " . __CLASS__ . PHP_EOL;
@@ -55,18 +54,379 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
      */
     public function testLeadEventsActivityGenerator()
     {
+        $events = [];
         $keyAliases = App_Constant_Table_SalesFunnelLeadActivityEvent::getKeyAlias();
-        echo "<pre>" . print_r($keyAliases, true); echo "</pre>"; // FrolovDEBUG
-        exit();
-//        foreach ($keyAliases AS $alias) {
-//            if ($alias == 'App\Event\SalesFunnel\Type\LeadCreation') {
-//                $subject = (new \App\Event\SalesFunnel\Subject\Lead())
-//                    ->setLeadId(null);
-//                $event = new App\Event\SalesFunnel\Type\LeadCreation($subject);
-//            }
-//            $this->eventService->trigger($event);
-//        }
+        $eventsSize = sizeof($keyAliases);
+        foreach ($keyAliases AS $alias) {
+
+            $result = match($alias) {
+                'App\\Event\\SalesFunnel\\Type\\LeadCreation' => function () {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new \App\Event\SalesFunnel\Type\LeadCreation($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadClosed' => function () {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\LeadClosed($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadFieldChanged' => function () {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadField())
+                        ->setLeadId(null)
+                        ->setField(null)
+                        ->setOldValue(null)
+                        ->setNewValue(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadFieldChanged($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadStageChanged' => function () {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadField())
+                        ->setLeadId(null)
+                        ->setField(null)
+                        ->setOldValue(null)
+                        ->setNewValue(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadStageChanged($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadClientChanged' => function () {
+                    return new App\Event\SalesFunnel\Type\LeadClientChanged(
+                        (new \App\Event\SalesFunnel\Subject\LeadField())
+                            ->setLeadId(null)
+                            ->setNewValue(null)
+                    );
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadOwnerChanged' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadField())
+                        ->setLeadId(null)
+                        ->setField(null)
+                        ->setOldValue(null)
+                        ->setNewValue(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadOwnerChanged($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadAssigneeChanged' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadField())
+                        ->setLeadId(null)
+                        ->setField(null)
+                        ->setOldValue(null)
+                        ->setNewValue(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadAssigneeChanged($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\QuantityControl\\LeadConfirmed' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadField())
+                        ->setLeadId(null)
+                        ->setField(null)
+                        ->setOldValue(null)
+                        ->setNewValue(null);
+
+
+                    return new App\Event\SalesFunnel\Type\QuantityControl\LeadConfirmed($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadPhoneAdded' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPhone())
+                        ->setLeadId(null)
+                        ->setLeadPhoneId(null)
+                        ->setPhone(null)
+                        ->setExt(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadPhoneAdded($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadPhoneRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPhone())
+                        ->setLeadId(null)
+                        ->setLeadPhoneId(null)
+                        ->setPhone(null)
+                        ->setExt(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadPhoneRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPhone())
+                        ->setLeadId(null)
+                        ->setLeadPhoneId(null)
+                        ->setPhone(null)
+                        ->setExt(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadEmailAdded' => function ()
+                {
+                    $subject = (new LeadEmail())
+                        ->setLeadId(null)
+                        ->setLeadEmailId(null)
+                        ->setEmail(null)
+                    ;
+                    return new App\Event\SalesFunnel\Type\LeadEmailAdded($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadEmailRemoved' => function ()
+                {
+                    $subject = (new LeadEmail())
+                        ->setLeadId(null)
+                        ->setLeadEmailId(null)
+                        ->setEmail(null)
+                    ;
+
+                    return new App\Event\SalesFunnel\Type\LeadEmailRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\SurveyAnswerAdded' => function ()
+                {
+                    $subject = (new LeadEmail())
+                        ->setLeadId(null)
+                        ->setLeadEmailId(null)
+                        ->setEmail(null)
+                    ;
+
+                    return new App\Event\SalesFunnel\Type\SurveyAnswerAdded($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadMergeStart' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadMerge())
+                        ->setTargetLeadId(null)
+                        ->setSourceLeadId(null)
+                        ->setReason(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadMergeStart($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\RelationWithCallCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPhone())
+                        ->setLeadId(null)
+                        ->setPhone(null)
+                    ;
+
+                    return new App\Event\SalesFunnel\Type\RelationWithCallCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\MissedCallCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPhone())
+                        ->setLeadId(null)
+                        ->setPhone(null)
+                    ;
+
+                    return new App\Event\SalesFunnel\Type\Relation\MissedCallCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\InvoiceCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\InvoiceCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\OfferCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\OfferCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CalculationSheetCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CalculationSheetCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\AgreementListCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\AgreementListCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ContractCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ContractCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ClaimCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ClaimCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ClientAddRequestCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ClientAddRequestCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CashAccountCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CashAccountCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CashMoneyCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CashMoneyCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\AvitoChatMessageCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanManufacturesCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanAccessoriesCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanRepairworksCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\EdnaMessageCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\EdnaMessageCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\RelationWithCallRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\RelationWithCallRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\InvoiceRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\InvoiceRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\OfferRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\OfferRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CalculationSheetRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CalculationSheetRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\AgreementListRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\AgreementListRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ContractRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ContractRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ClaimRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ClaimRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ClientAddRequestRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ClientAddRequestRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CashAccountRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CashAccountRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\CashMoneyRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\CashMoneyRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\AvitoChatMessageRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\AvitoChatMessageRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanManufacturesRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanManufacturesRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanAccessoriesRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanAccessoriesRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanRepairworksRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\ReceiptPlanRepairworksRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Relation\\MissedCallDeleted' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Relation\MissedCallDeleted($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\RelationWithMailServerMessageCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\RelationWithMailServerMessageCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\RelationWithMailServerMessageRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\RelationWithMailServerMessageRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\ChildLeadCreated' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\ChildLead())
+                        ->setLeadId(null)
+                        ->setParentLeadId(null);
+
+
+                    return new App\Event\SalesFunnel\Type\ChildLeadCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\ChildLeadRemoved' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\ChildLeadRemoved($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\File\\Added' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadFile())
+                        ->setLeadId(null)
+                        ->setFileId(null)
+                        ->setFileName(null);
+
+                    return new App\Event\SalesFunnel\Type\File\Added($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\File\\Removed' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadFile())
+                        ->setLeadId(null)
+                        ->setFileId(null)
+                        ->setFileName(null);
+
+                    return new App\Event\SalesFunnel\Type\File\Removed($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\OwnershipTransfer\\RequestCreated' => function ()
+                {
+                    $subject = (new OwnershipTransferSubject())
+                        ->setRequestId(null);
+
+                    return new App\Event\SalesFunnel\Type\OwnershipTransfer\RequestCreated($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Channel\\SentChannelMessageEvent' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Channel\SentChannelMessageEvent($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\Channel\\FailedChannelMessageEvent' => function ()
+                {
+                    $subject = (new \App\Event\SalesFunnel\Subject\Lead())->setLeadId(null);
+                    return new App\Event\SalesFunnel\Type\Channel\FailedChannelMessageEvent($subject);
+                },
+                'App\\Event\\SalesFunnel\\Type\\LeadPropValueChanged' => function () {
+                    $subject = (new \App\Event\SalesFunnel\Subject\LeadPropValue())
+                        ->setLeadId(null);
+
+                    return new App\Event\SalesFunnel\Type\LeadPropValueChanged($subject);
+                },
+                default => throw new \InvalidArgumentException("Неизвестный alias: $alias"),
+
+
+            };
+
         echo "inside test " . __METHOD__ . PHP_EOL;
 
     }
+    }
 }
\ No newline at end of file
