diff --git a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
index 425c26ab26..4e6016beb6 100644
--- a/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
+++ b/tests/suites/Integration/library/App/SalesFunnel/EventGenerationTest.php
@@ -88,10 +88,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
      */
     private ?array $initialLeadActivityFields = [];
 
-    /**
-     * @var bool
-     */
-    private bool $assertionType = true;
+    private $assertionType;
 
     /**
      * @var bool
@@ -103,6 +100,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
      */
     protected array $createdEvents = [];
 
+    protected int $leadId = 0;
     /**
      * @return void
      * @throws Zend_Db_Statement_Exception
@@ -116,7 +114,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
 
         $this->leadPermission = $container->get(\App\SalesFunnel\Permission\LeadPermission::class);
         $this->funnelPermission = $container->get(\App\SalesFunnel\Permission\FunnelPermission::class);
-        $this->dealsRatingRepository = 
+        $this->dealsRatingRepository =
             $container->get(\App\SalesFunnel\LeadDealsRating\Repositories\DealsRatingRepository::class);
         $this->requestFactory = $container->get(App\SalesFunnel\LeadOwnershipTransfer\Services\Factory::class);
         $this->requestRepository = $container->get(App\SalesFunnel\LeadOwnershipTransfer\Repository::class);
@@ -140,7 +138,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
 
         $subscriber = new LeadActivitySubscriber($this->leadFactoryMock, $this->leadCrudMock);
 
-        $this->eventService->setImmediateSubscriberClassList([]);
+        $this->eventService->setImmediateSubscriberClassList([$subscriber::class]);
         $this->eventService->getDispatcher()->addSubscriber($subscriber);
         authAsSystem();
 
@@ -183,8 +181,27 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
 
         $this->leadFactoryMock->setLeadModel($lead);
         $this->assertionType = true;
-        $this->actual = true; // сброс перед запуском
-        $this->eventGeneration($events, $fields, $lead);
+        $this->actual = true;
+        $this->eventGeneration($events, $fields, $lead, $this->assertionType);
+    }
+
+    public function testLeadDisabledEventsActivityGenerator()
+    {
+        $this->setEventsToInActive();
+        $this->setFieldsToInActive();
+        $events = $this->getAuditableInActiveEvents();
+        $fields = $this->getAuditableInActiveFields();
+        $totalAssertions = $this->getTotalExpectedAssertionsOfInActiveEvents();
+
+        echo "Ожидается $totalAssertions проверок " . PHP_EOL;
+
+        $this->lead = $this->leadCreation();
+        $this->bindEventsAssertions($this->lead);
+
+        $this->leadFactoryMock->setLeadModel($this->lead);
+        $this->assertionType = false;
+        $this->actual = false;
+        $this->eventGeneration($events, $fields, $this->lead, $this->assertionType);
     }
 
     /**
@@ -218,6 +235,13 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         return count($this->getAuditableActiveEvents()) + count($this->getAuditableActiveFields()) - 1;
     }
 
+    /**
+     * @return int
+     */
+    private function getTotalExpectedAssertionsOfInActiveEvents(): int
+    {
+        return count($this->getAuditableInActiveEvents()) + count($this->getAuditableInActiveFields()) - 1;
+    }
     /**
      * @return array
      * @throws Zend_Db_Statement_Exception
@@ -229,6 +253,21 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         ]);
     }
 
+    /**
+     * @return array
+     * @throws Zend_Db_Statement_Exception
+     */
+    private function getAuditableInActiveEvents(): array
+    {
+        $eventName = App_Db::get()->quote(App\Event\SalesFunnel\Type\LeadFieldChanged::class);
+        return App_Db_SalesFunnelLeadActivityEvent::obtain()->getRowsField('event_name', [
+            'id IN (?)' => new Zend_Db_Expr("
+                SELECT id
+                FROM sales_funnel_lead_activity_event
+                WHERE is_active = 0 OR (event_name = {$eventName}) 
+            ")
+        ]);
+    }
 
     /**
      * @return array
@@ -241,6 +280,17 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         ]);
     }
 
+    /**
+     * @return array
+     * @throws Zend_Db_Statement_Exception
+     */
+    private function getAuditableInActiveFields(): array
+    {
+        return App_Db_SalesFunnelLeadActivityField::obtain()->getRowsField('field', [
+            'is_active = ?' => 0
+        ]);
+    }
+
     /**
      * @return void
      */
@@ -249,6 +299,17 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         App_Db_SalesFunnelLeadActivityEvent::obtain()->update(['is_active' => 1], ['id > ?' => 0]);
     }
 
+    /**
+     * @return void
+     */
+    private function setEventsToInActive(): void
+    {
+        App_Db_SalesFunnelLeadActivityEvent::obtain()->update(['is_active' => 0], ['id > ?' => 0]);
+        App_Db_SalesFunnelLeadActivityEvent::obtain()->update(['is_active' => 1], [
+            'event_name = ?' => App\Event\SalesFunnel\Type\LeadFieldChanged::class
+        ]);
+    }
+
     /**
      * @return void
      */
@@ -257,6 +318,14 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         App_Db_SalesFunnelLeadActivityField::obtain()->update(['is_active' => 1], [], []);
     }
 
+    /**
+     * @return void
+     */
+    private function setFieldsToInActive(): void
+    {
+        App_Db_SalesFunnelLeadActivityField::obtain()->update(['is_active' => 0], [], []);
+    }
+
 
     /**
      * @param \App\SalesFunnel\Lead\Model $lead
@@ -264,11 +333,17 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
      */
     private function bindEventsAssertions(App\SalesFunnel\Lead\Model $lead): void
     {
-        $lead->getEntity()->eventBind(FieldChange::name(), function (FieldChange $event) {
-            if ($event->getField() === 'lastActivityDate') {
-                $this->actual = !$this->actual; // переключение состояния — только при реальном вызове
-            }
-        });
+        $realLead = $this->leadFactoryMock->fromId($lead->getEntity()->getId());
+
+        if ($this->assertionType) {
+            $realLead->getEntity()->eventBind(FieldChange::name(), function ($event) {
+                if ($event->getField() === 'lastActivityDate') {
+                    $this->actual = !$this->actual;
+                }
+            });
+        } else {
+            $this->actual = !$this->actual;
+        }
     }
 
     /**
@@ -280,9 +355,18 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
     public function eventGeneration(
         array                       $events,
         array                       $fields,
-        \App\SalesFunnel\Lead\Model $lead
+        \App\SalesFunnel\Lead\Model $lead,
+        bool                        $expectedResult
     )
     {
+        if ($expectedResult) {
+            $message         = "Событие %s не изменило lastActivityDate для лида %s, хотя должно было";
+            $messageForField = "Событие %s (поле: %s) не изменило lastActivityDate для лида %s, хотя должно было";
+        } else {
+            $message         = "Событие %s изменило lastActivityDate для лида %s, хотя не должно было";
+            $messageForField = "Событие %s (поле: %s) изменило lastActivityDate для лида %s, хотя не должно было";
+        }
+
         foreach ($events as $key => $alias) {
             $this->createdEvents[] = $alias;
             $eventGeneration = match ($alias) {
@@ -336,7 +420,7 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                     );
                 },
 
-                App\Event\SalesFunnel\Type\LeadFieldChanged::class => array_map(function ($field) use ($lead) {
+                App\Event\SalesFunnel\Type\LeadFieldChanged::class => array_map(function ($field) use ($lead, $message) {
                     return function () use (
                         $lead,
                         $field
@@ -350,6 +434,8 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
                                     ->setNewValue(null)
                             )
                         );
+
+                        return $field;
                     };
                 }, $fields),
 
@@ -852,23 +938,32 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
 
             if (!empty($eventGeneration)) {
                 if (is_callable($eventGeneration)) {
-                    $this->actual = !$this->assertionType;
+                    if ($this->assertionType) {
+//                        $this->actual = $this->assertionType;
+                    } else {
+//                        $this->actual = $this->assertionType;
+                    }
                     call_user_func($eventGeneration);
 
                     $this->assertEquals(
                         $this->assertionType, 
-                        $this->actual, 
-                        "Событие $alias изменило lastActivityDate, хотя не должно было"
+                        $this->actual,
+                        sprintf($message, $alias, $this->leadId)
                     );
 
                 } elseif (is_array($eventGeneration)) {
                     foreach ($eventGeneration as $event) {
-                        $this->actual = !$this->assertionType;
-                        call_user_func($event);
+                        if ($this->assertionType) {
+//                            $this->actual = !$this->assertionType;
+//                            $this->actual = $this->assertionType;
+                        } else {
+//                            $this->actual = $this->assertionType;
+                        }
+                        $field = call_user_func($event);
                         $this->assertEquals(
-                            $this->assertionType, 
+                            $this->assertionType,
                             $this->actual,
-                            "Событие $alias (поле) изменило lastActivityDate, хотя не должно было"
+                            sprintf($messageForField, $alias, $field, $this->leadId)
                         );
                     }
                 }
@@ -894,9 +989,11 @@ class EventGenerationTest extends \PHPUnit\Framework\TestCase
         $form->getElement('funnelId')->setValue(self::FUNNEL_ID);
         $form->getElement('plannedSum')->setValue(1.2);
 
-        return $this->creator->createLeadByUser(
+        $this->leadId = $this->creator->createLeadByUser(
             1160,
             $form
-        );
+        )->getEntity()->getId();
+
+        return $this->leadFactoryMock->fromId($this->leadId);
     }
 }
\ No newline at end of file
