diff --git a/application/configs/templatedb/tables/sales_funnel_lead_activity_field.ini b/application/configs/templatedb/tables/sales_funnel_lead_activity_field.ini
index e69de29bb2..cc9c1ffbb1 100644
--- a/application/configs/templatedb/tables/sales_funnel_lead_activity_field.ini
+++ b/application/configs/templatedb/tables/sales_funnel_lead_activity_field.ini
@@ -0,0 +1,32 @@
+;конфигурация для таблицы 0 - обычная, 1 = каталог
+[config]
+catalog = 1
+searchkey = "id"
+
+; если кто-то будет делать из таблицы каталог, знайте что поле is_active не должно синхронизироваться
+[exceptions]
+ex1 = is_active
+
+; Секция constant - содержит настройки для генерации файлов по каталожным таблицам
+[constant]
+; Константы
+constant.key = 'field'
+constant.value = 'id'
+constant.sort_field = 'field';
+constant.comment_field = 'field_title'
+constant.prefix = 'type_'
+; Ключ = заголовок
+keyTitle.key = 'id'
+keyTitle.value = 'field_title'
+keyTitle.sort_field = 'id'
+keyTitle.comment_field = 'field_title'
+; Ключ = алиас
+keyAlias.key = 'id'
+keyAlias.value = 'field';
+keyAlias.sort_field = 'id'
+keyAlias.comment_field = 'field_title'
+; Алиас = ключ
+aliasKey.key = 'key';
+aliasKey.value = 'field_title'
+aliasKey.sort_field = 'id'
+aliasKey.comment_field = 'type_'
\ No newline at end of file
diff --git a/application/modules/sales-funnel/controllers/SettingsController.php b/application/modules/sales-funnel/controllers/SettingsController.php
index 13cfa80a63..cc1160b015 100644
--- a/application/modules/sales-funnel/controllers/SettingsController.php
+++ b/application/modules/sales-funnel/controllers/SettingsController.php
@@ -73,6 +73,10 @@ class SalesFunnel_SettingsController extends Zend_Controller_Action
                 'is_active'
             ]);
 
+            $fields = App_Db_SalesFunnelLeadActivityField::obtain()->getRowsFields(
+                ['id','field', 'field_title'],[],[], Zend_Db::FETCH_ASSOC
+            );
+
             return [
                 'result' => [
                     'settings' => $this->settingsMapper->getSettings(),
@@ -83,7 +87,8 @@ class SalesFunnel_SettingsController extends Zend_Controller_Action
                         $this->leadPropCategoryFactory->all()->getItems()
                     ),
                     'dealsRating' => $this->leadDealsRatingRepository->getAll(),
-                    'eventActivities' => $events
+                    'eventActivities' => $events,
+                    'fields' => $fields
                 ],
             ];
         });
diff --git a/library/App/CodeGenerator/ClassGenerator/DbTable/Builder.php b/library/App/CodeGenerator/ClassGenerator/DbTable/Builder.php
index 32d7c64569..210550a0be 100644
--- a/library/App/CodeGenerator/ClassGenerator/DbTable/Builder.php
+++ b/library/App/CodeGenerator/ClassGenerator/DbTable/Builder.php
@@ -33,6 +33,7 @@ class Builder
 
     public function generate()
     {
+        $this->_options['creator'] = 'Frolov Anatoliyfrolov_anatoliy@amd-co.ru>';
         if (empty($this->_options['creator'])) {
             throw new \RuntimeException('Укажите автора');
         }
diff --git a/library/App/Constant/Table/SalesFunnelLeadActivityField.php b/library/App/Constant/Table/SalesFunnelLeadActivityField.php
index c1952a64bf..829fb4d556 100644
--- a/library/App/Constant/Table/SalesFunnelLeadActivityField.php
+++ b/library/App/Constant/Table/SalesFunnelLeadActivityField.php
@@ -7,94 +7,94 @@ class App_Constant_Table_SalesFunnelLeadActivityField extends App_Constant_Table
 {
 
     /**
-     * 
+     * Владелец
      */
-    const  = 1;
+    const TYPE_OWNER = 1;
 
     /**
-     * 
+     * Исполнитель
      */
-    const  = 2;
+    const TYPE_ASSIGNEE = 2;
 
     /**
-     * 
+     * Воронка
      */
-    const  = 3;
+    const TYPE_FUNNEL = 3;
 
     /**
-     * 
+     * Этап
      */
-    const  = 4;
+    const TYPE_STAGE = 4;
 
     /**
-     * 
+     * Название
      */
-    const  = 5;
+    const TYPE_NAME = 5;
 
     /**
-     * 
+     * Телефоны
      */
-    const  = 6;
+    const TYPE_PHONES = 6;
 
     /**
-     * 
+     * Почты
      */
-    const  = 7;
+    const TYPE_EMAILS = 7;
 
     /**
-     * 
+     * Клиент
      */
-    const  = 8;
+    const TYPE_CLIENT = 8;
 
     /**
-     * 
+     * ИНН клиента
      */
-    const  = 9;
+    const TYPE_CLIENT_INN = 9;
 
     /**
-     * 
+     * Примечание по лиду
      */
-    const  = 10;
+    const TYPE_ANNOTATION = 10;
 
     /**
-     * 
+     * Дата планируемой сделки
      */
-    const  = 11;
+    const TYPE_PLANNED_DATE = 11;
 
     /**
-     * 
+     * Планируемая сумма
      */
-    const  = 12;
+    const TYPE_PLANNED_SUM = 12;
 
     /**
-     * 
+     * Фактическая сумма
      */
-    const  = 13;
+    const TYPE_ACTUAL_SUM = 13;
 
     /**
-     * 
+     * Файлы
      */
-    const  = 14;
+    const TYPE_FILES = 14;
 
     /**
      * Ключ = заголовок
      * @var array
      */
     protected static $_keyTitle = array(
-        1 => '', // 
-        2 => '', // 
-        3 => '', // 
-        4 => '', // 
-        5 => '', // 
-        6 => '', // 
-        7 => '', // 
-        8 => '', // 
-        9 => '', // 
-        10 => '', // 
-        11 => '', // 
-        12 => '', // 
-        13 => '', // 
-        14 => '', // 
+        1 => 'Владелец', // Владелец
+        2 => 'Исполнитель', // Исполнитель
+        3 => 'Воронка', // Воронка
+        4 => 'Этап', // Этап
+        5 => 'Название', // Название
+        6 => 'Телефоны', // Телефоны
+        7 => 'Почты', // Почты
+        8 => 'Клиент', // Клиент
+        9 => 'ИНН клиента', // ИНН клиента
+        10 => 'Примечание по лиду', // Примечание по лиду
+        11 => 'Дата планируемой сделки', // Дата планируемой сделки
+        12 => 'Планируемая сумма', // Планируемая сумма
+        13 => 'Фактическая сумма', // Фактическая сумма
+        14 => 'Файлы', // Файлы
     );
 
     /**
@@ -102,20 +102,20 @@ class App_Constant_Table_SalesFunnelLeadActivityField extends App_Constant_Table
      * @var array
      */
     protected static $_keyAlias = array(
-        1 => '', // 
-        2 => '', // 
-        3 => '', // 
-        4 => '', // 
-        5 => '', // 
-        6 => '', // 
-        7 => '', // 
-        8 => '', // 
-        9 => '', // 
-        10 => '', // 
-        11 => '', // 
-        12 => '', // 
-        13 => '', // 
-        14 => '', // 
+        1 => 'owner', // Владелец
+        2 => 'assignee', // Исполнитель
+        3 => 'funnel', // Воронка
+        4 => 'stage', // Этап
+        5 => 'name', // Название
+        6 => 'phones', // Телефоны
+        7 => 'emails', // Почты
+        8 => 'client', // Клиент
+        9 => 'client_inn', // ИНН клиента
+        10 => 'annotation', // Примечание по лиду
+        11 => 'planned_date', // Дата планируемой сделки
+        12 => 'planned_sum', // Планируемая сумма
+        13 => 'actual_sum', // Фактическая сумма
+        14 => 'files', // Файлы
     );
 
     /**
@@ -123,19 +123,19 @@ class App_Constant_Table_SalesFunnelLeadActivityField extends App_Constant_Table
      * @var array
      */
     protected static $_aliasKey = array(
-        '' => 1, // 
-        '' => 2, // 
-        '' => 3, // 
-        '' => 4, // 
-        '' => 5, // 
-        '' => 6, // 
-        '' => 7, // 
-        '' => 8, // 
-        '' => 9, // 
-        '' => 10, // 
-        '' => 11, // 
-        '' => 12, // 
-        '' => 13, // 
-        '' => 14, // 
+        '' => 'Владелец', // 
+        '' => 'Исполнитель', // 
+        '' => 'Воронка', // 
+        '' => 'Этап', // 
+        '' => 'Название', // 
+        '' => 'Телефоны', // 
+        '' => 'Почты', // 
+        '' => 'Клиент', // 
+        '' => 'ИНН клиента', // 
+        '' => 'Примечание по лиду', // 
+        '' => 'Дата планируемой сделки', // 
+        '' => 'Планируемая сумма', // 
+        '' => 'Фактическая сумма', // 
+        '' => 'Файлы', // 
     );
 }
diff --git a/library/App/Constant/TableList.php b/library/App/Constant/TableList.php
index 70c2d5af65..430a9b10e2 100644
--- a/library/App/Constant/TableList.php
+++ b/library/App/Constant/TableList.php
@@ -261,12 +261,12 @@ define('DB_ANALYTICS_PROFITABILITY_VARS', 'analytics_profitability_vars');
 define('DB_ANALYTICS_PROJECT_CAPITALIZATION', 'analytics_project_capitalization');
 
 /**
- * Анализ продаж, исходные данные для отчета
+ * ToDo добавить комментарий
  */
 define('DB_ANALYTICS_SALES_ANALYSIS', 'analytics_sales_analysis');
 
 /**
- * Анализ продаж, маржинальность по периодам (создается в коде)
+ * ToDo добавить комментарий
  */
 define('DB_ANALYTICS_SALES_ANALYSIS_MARGINALITY', 'analytics_sales_analysis_marginality');
 
@@ -276,7 +276,7 @@ define('DB_ANALYTICS_SALES_ANALYSIS_MARGINALITY', 'analytics_sales_analysis_marg
 define('DB_ANALYTICS_SALES_ANALYSIS_QUEUE', 'analytics_sales_analysis_queue');
 
 /**
- * Анализ продаж, оборот по периодам (создается в коде)
+ * ToDo добавить комментарий
  */
 define('DB_ANALYTICS_SALES_ANALYSIS_TURNOVER', 'analytics_sales_analysis_turnover');
 
@@ -631,7 +631,7 @@ define('DB_CALL_RECORDS', 'call_records');
 define('DB_CALL_REGISTRY', 'call_registry');
 
 /**
- * анализ звонков: настройки анализа
+ * ToDo добавить комментарий
  */
 define('DB_CALL_REGISTRY_ANALYSIS_SETTINGS', 'call_registry_analysis_settings');
 
@@ -791,7 +791,7 @@ define('DB_CASH_DIRECTOR_WAGE_EXCEPTIONS', 'cash_director_wage_exceptions');
 define('DB_CASH_DUPLICATES', 'cash_duplicates');
 
 /**
- * Статьи расходов
+ * Статья расхода. ВНИМАНИЕ! При изменении данных в таблице необходимо также внести изменения на в cash_expense_items на проекте ВМК
  */
 define('DB_CASH_EXPENSE_ITEMS', 'cash_expense_items');
 
@@ -1686,7 +1686,7 @@ define('DB_CLIENT_LIST_AGREEMENT_CLIENT_CHECK', 'client_list_agreement_client_ch
 define('DB_CLIENT_LIST_AGREEMENT_CONTRACTS', 'client_list_agreement_contracts');
 
 /**
- * таблица принятых решений в листе согласования
+ * таблица принятых решений по связи листа согласования с шаблоном договора
  */
 define('DB_CLIENT_LIST_AGREEMENT_DECISIONS', 'client_list_agreement_decisions');
 
@@ -2031,12 +2031,12 @@ define('DB_CORP_PHONE_TARIFF', 'corp_phone_tariff');
 define('DB_COUNTRY', 'country');
 
 /**
- * Список кронов
+ * ToDo добавить комментарий
  */
 define('DB_CRON_LIST', 'cron_list');
 
 /**
- * Список кронов по проектам
+ * ToDo добавить комментарий
  */
 define('DB_CRON_LIST_PROJECTS', 'cron_list_projects');
 
@@ -4131,7 +4131,7 @@ define('DB_RECEIPT_PRICE_TOOL_GROUPS', 'receipt_price_tool_groups');
 define('DB_RECEIPT_PURCHASES', 'receipt_purchases');
 
 /**
- * Необходимые закупки детализация
+ * ToDo добавить комментарий
  */
 define('DB_RECEIPT_PURCHASES_DETAILS', 'receipt_purchases_details');
 
@@ -4246,7 +4246,7 @@ define('DB_RECEIPT_TREE', 'receipt_tree');
 define('DB_RECEIPT_TREE_TEMPLATE', 'receipt_tree_template');
 
 /**
- * rechka ai: список отчетов
+ * ToDo добавить комментарий
  */
 define('DB_RECHKAAI_REPORT', 'rechkaai_report');
 
@@ -4271,7 +4271,7 @@ define('DB_RECHKAAI_TASK_CALLREPORT_FIELD_TYPE', 'rechkaai_task_callreport_field
 define('DB_RECHKAAI_TASK_CALLREPORT_GROUP', 'rechkaai_task_callreport_group');
 
 /**
- * rechka ai: правила создания заданий и настройки заданий
+ * ToDo добавить комментарий
  */
 define('DB_RECHKAAI_TASK_CREATION_RULE', 'rechkaai_task_creation_rule');
 
@@ -4756,10 +4756,15 @@ define('DB_SALES_FUNNEL_FILTER_USER', 'sales_funnel_filter_user');
 define('DB_SALES_FUNNEL_LEAD', 'sales_funnel_lead');
 
 /**
- * Указания какие события являются активностями
+ * Воронка продаж: Указания какие события являются активностями
  */
 define('DB_SALES_FUNNEL_LEAD_ACTIVITY_EVENT', 'sales_funnel_lead_activity_event');
 
+/**
+ * Воронка продаж: настройка учёта изменений полей лида как активности
+ */
+define('DB_SALES_FUNNEL_LEAD_ACTIVITY_FIELD', 'sales_funnel_lead_activity_field');
+
 /**
  * Агрегированные данные лидов
  */
@@ -5936,6 +5941,6 @@ define('DB_YANDEX_360_ORGANIZATION', 'yandex_360_organization');
 define('DB_YANDEX_360_ORGANIZATION_DOMAIN', 'yandex_360_organization_domain');
 
 /**
- * Аккаунты Zoho
+ * ToDo добавить комментарий
  */
 define('DB_ZOHO_ACCOUNT', 'zoho_account');
diff --git a/library/App/SalesFunnel/Form/SettingsForm.php b/library/App/SalesFunnel/Form/SettingsForm.php
index c0c52c33bc..3acd10e9bc 100644
--- a/library/App/SalesFunnel/Form/SettingsForm.php
+++ b/library/App/SalesFunnel/Form/SettingsForm.php
@@ -281,6 +281,9 @@ class SettingsForm extends App_Form
                     (new Zend_Form_Element('is_active'))
                         ->setRequired()
                 )
+                ->addElement(
+                    (new Zend_Form_Element('additional'))
+                )
         );
     }
 }
diff --git a/library/App/SalesFunnel/Settings/HttpHandler.php b/library/App/SalesFunnel/Settings/HttpHandler.php
index 85310c6d38..096d9febca 100644
--- a/library/App/SalesFunnel/Settings/HttpHandler.php
+++ b/library/App/SalesFunnel/Settings/HttpHandler.php
@@ -96,6 +96,8 @@ class HttpHandler
 
         $this->settingsMapper->saveSettings($settings);
 
+        echo "<pre>" . print_r($form->getValue('eventActivitiesSettings'), true); echo "</pre>"; // FrolovDEBUG
+        exit();
         $this->saveDealsRating($form->getValue('dealsRating'));
         $this->setEventActivityMarker($form->getValue('eventActivitiesSettings'));
 
diff --git a/public/js/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.js b/public/js/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.js
index f3423ef669..c5ed3d3484 100644
--- a/public/js/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.js
+++ b/public/js/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.js
@@ -10,52 +10,52 @@ define(["require", "exports", "../../../decorator/module.decorator"], function (
     exports.ConstantTableSalesFunnelLeadActivityFieldModule = exports.ALIAS = exports.TITLES = exports.EConstantTableSalesFunnelLeadActivityField = void 0;
     var EConstantTableSalesFunnelLeadActivityField;
     (function (EConstantTableSalesFunnelLeadActivityField) {
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 1] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 2] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 3] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 4] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 5] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 6] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 7] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 8] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 9] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 10] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 11] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 12] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 13] = "TYPE_";
-        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_"] = 14] = "TYPE_";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_OWNER"] = 1] = "TYPE_OWNER";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_ASSIGNEE"] = 2] = "TYPE_ASSIGNEE";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_FUNNEL"] = 3] = "TYPE_FUNNEL";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_STAGE"] = 4] = "TYPE_STAGE";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_NAME"] = 5] = "TYPE_NAME";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_PHONES"] = 6] = "TYPE_PHONES";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_EMAILS"] = 7] = "TYPE_EMAILS";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_CLIENT"] = 8] = "TYPE_CLIENT";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_CLIENT_INN"] = 9] = "TYPE_CLIENT_INN";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_ANNOTATION"] = 10] = "TYPE_ANNOTATION";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_PLANNED_DATE"] = 11] = "TYPE_PLANNED_DATE";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_PLANNED_SUM"] = 12] = "TYPE_PLANNED_SUM";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_ACTUAL_SUM"] = 13] = "TYPE_ACTUAL_SUM";
+        EConstantTableSalesFunnelLeadActivityField[EConstantTableSalesFunnelLeadActivityField["TYPE_FILES"] = 14] = "TYPE_FILES";
     })(EConstantTableSalesFunnelLeadActivityField || (exports.EConstantTableSalesFunnelLeadActivityField = EConstantTableSalesFunnelLeadActivityField = {}));
     exports.TITLES = {
-        '1': '',
-        '2': '',
-        '3': '',
-        '4': '',
-        '5': '',
-        '6': '',
-        '7': '',
-        '8': '',
-        '9': '',
-        '10': '',
-        '11': '',
-        '12': '',
-        '13': '',
-        '14': '',
+        '1': 'Владелец',
+        '2': 'Исполнитель',
+        '3': 'Воронка',
+        '4': 'Этап',
+        '5': 'Название',
+        '6': 'Телефоны',
+        '7': 'Почты',
+        '8': 'Клиент',
+        '9': 'ИНН клиента',
+        '10': 'Примечание по лиду',
+        '11': 'Дата планируемой сделки',
+        '12': 'Планируемая сумма',
+        '13': 'Фактическая сумма',
+        '14': 'Файлы',
     };
     exports.ALIAS = {
-        '1': '',
-        '2': '',
-        '3': '',
-        '4': '',
-        '5': '',
-        '6': '',
-        '7': '',
-        '8': '',
-        '9': '',
-        '10': '',
-        '11': '',
-        '12': '',
-        '13': '',
-        '14': '',
+        '1': 'owner',
+        '2': 'assignee',
+        '3': 'funnel',
+        '4': 'stage',
+        '5': 'name',
+        '6': 'phones',
+        '7': 'emails',
+        '8': 'client',
+        '9': 'client_inn',
+        '10': 'annotation',
+        '11': 'planned_date',
+        '12': 'planned_sum',
+        '13': 'actual_sum',
+        '14': 'files',
     };
     let ConstantTableSalesFunnelLeadActivityFieldModule = class ConstantTableSalesFunnelLeadActivityFieldModule {
     };
diff --git a/public/js/app/application/sales-funnel/funnel/component/funnel-settings.component.js b/public/js/app/application/sales-funnel/funnel/component/funnel-settings.component.js
index 4eaed00dd5..3adb8b9cc9 100644
--- a/public/js/app/application/sales-funnel/funnel/component/funnel-settings.component.js
+++ b/public/js/app/application/sales-funnel/funnel/component/funnel-settings.component.js
@@ -7,7 +7,7 @@ var __decorate = (this && this.__decorate) || function (decorators, target, key,
 var __param = (this && this.__param) || function (paramIndex, decorator) {
     return function (target, key) { decorator(target, key, paramIndex); }
 };
-define(["require", "exports", "../../../../angular/decorator/component.decorator", "../service/backend.service", "../../../../core/service/message.service", "../../../../core/service/toast.service", "../../../../angular/component/amd-switch/amd-switch.component"], function (require, exports, component_decorator_1, backend_service_1, message_service_1, toast_service_1, amd_switch_component_1) {
+define(["require", "exports", "../../../../angular/decorator/component.decorator", "../service/backend.service", "../../../../core/service/message.service", "../../../../core/service/toast.service", "../../../../angular/component/amd-switch/amd-switch.component", "../../../../core/model/data-source/array.data-source", "../../../../angular/component/amd-select/amd-select.filter"], function (require, exports, component_decorator_1, backend_service_1, message_service_1, toast_service_1, amd_switch_component_1, array_data_source_1, amd_select_filter_1) {
     "use strict";
     Object.defineProperty(exports, "__esModule", { value: true });
     exports.FunnelSettingsComponent = void 0;
@@ -59,6 +59,13 @@ define(["require", "exports", "../../../../angular/decorator/component.decorator
                             >
                                 {{item.title}}
                             </amd-switch>
+                            <span ng-if="item.id == 3">
+                                <amd-multiselect
+                                    ng-if="$ctrl.fieldsSource"
+                                    ng-model="$ctrl.fileds"
+                                    data-source="$ctrl.fieldsSource"
+                                ></amd-multiselect>
+                            </span>
                         </div>
                     </amd-zend-control>
                 </div>
@@ -91,6 +98,13 @@ define(["require", "exports", "../../../../angular/decorator/component.decorator
                     .map((item) => {
                     return { id: item.id, title: item.event_description, is_active: +item.is_active > 0 };
                 });
+                this.fields = data.fields;
+                this.fieldsSource = new array_data_source_1.default(new amd_select_filter_1.default(), Object.entries(this.fields).map(([key, value]) => {
+                    return {
+                        text: value.field_title,
+                        value: +key
+                    };
+                }));
             }, reason => (new message_service_1.default()).alert(reason));
         }
         save() {
@@ -99,7 +113,7 @@ define(["require", "exports", "../../../../angular/decorator/component.decorator
                 propCategories: this.propCategories,
                 dealsRating: this.dealsRating,
                 eventActivitiesSettings: this.eventActivitiesSettings.map(item => {
-                    return { id: item.id, is_active: item.is_active };
+                    return { id: item.id, is_active: item.is_active, additional: [] };
                 })
             })
                 .then(() => this.amdWindow.hide(true), reason => {
diff --git a/public/js/constant/table/salesfunnelleadactivityfield.js b/public/js/constant/table/salesfunnelleadactivityfield.js
index 1ffaf4b787..271a672e23 100644
--- a/public/js/constant/table/salesfunnelleadactivityfield.js
+++ b/public/js/constant/table/salesfunnelleadactivityfield.js
@@ -14,74 +14,74 @@ var Constant_Table_SalesFunnelLeadActivityField = (function (_super) {
         _super.apply(this, arguments);
 
         /**
-         * 
+         * Владелец
          */
-        this. = 1;
+        this.TYPE_OWNER = 1;
 
         /**
-         * 
+         * Исполнитель
          */
-        this. = 2;
+        this.TYPE_ASSIGNEE = 2;
 
         /**
-         * 
+         * Воронка
          */
-        this. = 3;
+        this.TYPE_FUNNEL = 3;
 
         /**
-         * 
+         * Этап
          */
-        this. = 4;
+        this.TYPE_STAGE = 4;
 
         /**
-         * 
+         * Название
          */
-        this. = 5;
+        this.TYPE_NAME = 5;
 
         /**
-         * 
+         * Телефоны
          */
-        this. = 6;
+        this.TYPE_PHONES = 6;
 
         /**
-         * 
+         * Почты
          */
-        this. = 7;
+        this.TYPE_EMAILS = 7;
 
         /**
-         * 
+         * Клиент
          */
-        this. = 8;
+        this.TYPE_CLIENT = 8;
 
         /**
-         * 
+         * ИНН клиента
          */
-        this. = 9;
+        this.TYPE_CLIENT_INN = 9;
 
         /**
-         * 
+         * Примечание по лиду
          */
-        this. = 10;
+        this.TYPE_ANNOTATION = 10;
 
         /**
-         * 
+         * Дата планируемой сделки
          */
-        this. = 11;
+        this.TYPE_PLANNED_DATE = 11;
 
         /**
-         * 
+         * Планируемая сумма
          */
-        this. = 12;
+        this.TYPE_PLANNED_SUM = 12;
 
         /**
-         * 
+         * Фактическая сумма
          */
-        this. = 13;
+        this.TYPE_ACTUAL_SUM = 13;
 
         /**
-         * 
+         * Файлы
          */
-        this. = 14;
+        this.TYPE_FILES = 14;
 
 
         /**
@@ -89,20 +89,20 @@ var Constant_Table_SalesFunnelLeadActivityField = (function (_super) {
          * @type {Object}
          */
         this._keyTitle = {
-            '1' : '', // 
-            '2' : '', // 
-            '3' : '', // 
-            '4' : '', // 
-            '5' : '', // 
-            '6' : '', // 
-            '7' : '', // 
-            '8' : '', // 
-            '9' : '', // 
-            '10' : '', // 
-            '11' : '', // 
-            '12' : '', // 
-            '13' : '', // 
-            '14' : '', // 
+            '1' : 'Владелец', // Владелец
+            '2' : 'Исполнитель', // Исполнитель
+            '3' : 'Воронка', // Воронка
+            '4' : 'Этап', // Этап
+            '5' : 'Название', // Название
+            '6' : 'Телефоны', // Телефоны
+            '7' : 'Почты', // Почты
+            '8' : 'Клиент', // Клиент
+            '9' : 'ИНН клиента', // ИНН клиента
+            '10' : 'Примечание по лиду', // Примечание по лиду
+            '11' : 'Дата планируемой сделки', // Дата планируемой сделки
+            '12' : 'Планируемая сумма', // Планируемая сумма
+            '13' : 'Фактическая сумма', // Фактическая сумма
+            '14' : 'Файлы', // Файлы
         };
 
 
@@ -111,20 +111,20 @@ var Constant_Table_SalesFunnelLeadActivityField = (function (_super) {
          * @type {Object}
          */
         this._keyAlias = {
-            '1' : '', // 
-            '2' : '', // 
-            '3' : '', // 
-            '4' : '', // 
-            '5' : '', // 
-            '6' : '', // 
-            '7' : '', // 
-            '8' : '', // 
-            '9' : '', // 
-            '10' : '', // 
-            '11' : '', // 
-            '12' : '', // 
-            '13' : '', // 
-            '14' : '', // 
+            '1' : 'owner', // Владелец
+            '2' : 'assignee', // Исполнитель
+            '3' : 'funnel', // Воронка
+            '4' : 'stage', // Этап
+            '5' : 'name', // Название
+            '6' : 'phones', // Телефоны
+            '7' : 'emails', // Почты
+            '8' : 'client', // Клиент
+            '9' : 'client_inn', // ИНН клиента
+            '10' : 'annotation', // Примечание по лиду
+            '11' : 'planned_date', // Дата планируемой сделки
+            '12' : 'planned_sum', // Планируемая сумма
+            '13' : 'actual_sum', // Фактическая сумма
+            '14' : 'files', // Файлы
         };
 
 
@@ -133,20 +133,20 @@ var Constant_Table_SalesFunnelLeadActivityField = (function (_super) {
          * @type {Object}
          */
         this._aliasKey = {
-            '' : 1, // 
-            '' : 2, // 
-            '' : 3, // 
-            '' : 4, // 
-            '' : 5, // 
-            '' : 6, // 
-            '' : 7, // 
-            '' : 8, // 
-            '' : 9, // 
-            '' : 10, // 
-            '' : 11, // 
-            '' : 12, // 
-            '' : 13, // 
-            '' : 14, // 
+            '' : 'Владелец', // 
+            '' : 'Исполнитель', // 
+            '' : 'Воронка', // 
+            '' : 'Этап', // 
+            '' : 'Название', // 
+            '' : 'Телефоны', // 
+            '' : 'Почты', // 
+            '' : 'Клиент', // 
+            '' : 'ИНН клиента', // 
+            '' : 'Примечание по лиду', // 
+            '' : 'Дата планируемой сделки', // 
+            '' : 'Планируемая сумма', // 
+            '' : 'Фактическая сумма', // 
+            '' : 'Файлы', // 
         };
     }
 
diff --git a/query/admn-376.sql b/query/admn-376.sql
index 3235ff6c5f..8300f98fbe 100644
--- a/query/admn-376.sql
+++ b/query/admn-376.sql
@@ -1,20 +1,41 @@
 DROP TABLE IF EXISTS `sales_funnel_lead_activity_event`;
-
 CREATE TABLE `sales_funnel_lead_activity_event` (
     `id` INT AUTO_INCREMENT PRIMARY KEY,
     `event_name` VARCHAR(150) NOT NULL COMMENT 'Название события',
     `event_description` VARCHAR(1000) COMMENT 'Описание события',
     `is_active` TINYINT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'считается ли это событие активностью',
-    `options` JSON COMMENT 'Дополнительные опции',
     INDEX (event_name)
 ) COMMENT='Воронка продаж: Указания какие события являются активностями';
 
 
+CREATE TABLE `sales_funnel_lead_activity_field` (
+       `id` INT AUTO_INCREMENT PRIMARY KEY,
+       `field` VARCHAR(50) NOT NULL COMMENT 'field, поле в лиде',
+       `field_title` VARCHAR(50) NOT NULL COMMENT 'Понятное человеку название поля лида',
+       `is_active` TINYINT UNSIGNED NOT NULL DEFAULT 1 COMMENT 'считается ли изменение этого поля активностью',
+       UNIQUE KEY `field` (`field`)
+) COMMENT = 'Воронка продаж: настройка учёта изменений полей лида как активности';
+
+
+INSERT INTO `sales_funnel_lead_activity_field` (`field`, `field_title`)
+VALUES
+    ('owner', 'Владелец'),
+    ('assignee', 'Исполнитель'),
+    ('funnel', 'Воронка'),
+    ('stage', 'Этап'),
+    ('name', 'Название'),
+    ('client', 'Клиент'),
+    ('client_inn', 'ИНН клиента'),
+    ('annotation', 'Примечание по лиду'),
+    ('planned_date', 'Дата планируемой сделки'),
+    ('planned_sum', 'Планируемая сумма'),
+    ('actual_sum', 'Фактическая сумма'),
+
+
 INSERT INTO `sales_funnel_lead_activity_event` (`event_name`, `event_description`)
 VALUES
     ('App\\Event\\SalesFunnel\\Type\\LeadCreation', 'Создание лида'),
     ('App\\Event\\SalesFunnel\\Type\\LeadClosed', 'Переход лида на закрытый этап воронки'),
-    ('App\\Event\\SalesFunnel\\Type\\LeadCommentAdded', 'Добавление комментария к лиду'),
     ('App\\Event\\SalesFunnel\\Type\\LeadFieldChanged', 'Изменение пользовательского поля лида'),
     ('App\\Event\\SalesFunnel\\Type\\LeadStageChanged', 'Изменение этапа воронки у лида'),
     ('App\\Event\\SalesFunnel\\Type\\LeadClientChanged', 'Изменение привязанного клиента у лида'),
@@ -39,8 +60,6 @@ VALUES
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ClientAddRequestCreated', 'Создание запроса на добавление клиента'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\CashAccountCreated', 'Создание кассового счёта'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\CashMoneyCreated', 'Фиксация движения денежных средств'),
-    ('App\\Event\\SalesFunnel\\Type\\Relation\\CalendarNotificationCreated', 'Создание календарного уведомления'),
-    ('App\\Event\\SalesFunnel\\Type\\Relation\\StaffTaskCreated', 'Создание задачи для сотрудника, связанной с лидом'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\AvitoChatMessageCreated', 'Получение сообщения из чата Avito, привязанного к лиду'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanManufacturesCreated', 'Создание связи с планом производства изделий'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanAccessoriesCreated', 'Создание связи с планом производства комплектующих'),
@@ -56,8 +75,6 @@ VALUES
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ClientAddRequestRemoved', 'Удаление запроса на добавление клиента'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\CashAccountRemoved', 'Удаление безнальной транзакции'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\CashMoneyRemoved', 'Удаление нальной транзакции'),
-    ('App\\Event\\SalesFunnel\\Type\\Relation\\CalendarNotificationRemoved', 'Удаление уведомления календаря'),
-    ('App\\Event\\SalesFunnel\\Type\\Relation\\StaffTaskRemoved', 'Удаление задачи для сотрудника'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\AvitoChatMessageRemoved', 'Удаление сообщения из чата Avito'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanManufacturesRemoved', 'Удаление связи с планом производства изделий'),
     ('App\\Event\\SalesFunnel\\Type\\Relation\\ReceiptPlanAccessoriesRemoved', 'Удаление связи с планом производства комплектующих'),
diff --git a/typescript/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.ts b/typescript/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.ts
index 1ab9dd6375..de8b6b0b0e 100644
--- a/typescript/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.ts
+++ b/typescript/app/angular/generated/constant/table/sales-funnel-lead-activity-field.module.ts
@@ -7,110 +7,110 @@ export enum EConstantTableSalesFunnelLeadActivityField
 {
 
         /**
-         * 
+         * Владелец
          */
-         = 1,
+        TYPE_OWNER = 1,
 
         /**
-         * 
+         * Исполнитель
          */
-         = 2,
+        TYPE_ASSIGNEE = 2,
 
         /**
-         * 
+         * Воронка
          */
-         = 3,
+        TYPE_FUNNEL = 3,
 
         /**
-         * 
+         * Этап
          */
-         = 4,
+        TYPE_STAGE = 4,
 
         /**
-         * 
+         * Название
          */
-         = 5,
+        TYPE_NAME = 5,
 
         /**
-         * 
+         * Телефоны
          */
-         = 6,
+        TYPE_PHONES = 6,
 
         /**
-         * 
+         * Почты
          */
-         = 7,
+        TYPE_EMAILS = 7,
 
         /**
-         * 
+         * Клиент
          */
-         = 8,
+        TYPE_CLIENT = 8,
 
         /**
-         * 
+         * ИНН клиента
          */
-         = 9,
+        TYPE_CLIENT_INN = 9,
 
         /**
-         * 
+         * Примечание по лиду
          */
-         = 10,
+        TYPE_ANNOTATION = 10,
 
         /**
-         * 
+         * Дата планируемой сделки
          */
-         = 11,
+        TYPE_PLANNED_DATE = 11,
 
         /**
-         * 
+         * Планируемая сумма
          */
-         = 12,
+        TYPE_PLANNED_SUM = 12,
 
         /**
-         * 
+         * Фактическая сумма
          */
-         = 13,
+        TYPE_ACTUAL_SUM = 13,
 
         /**
-         * 
+         * Файлы
          */
-         = 14,
+        TYPE_FILES = 14,
 }
 
 
 export const TITLES = {
-        '1': '', // ''
-        '2': '', // ''
-        '3': '', // ''
-        '4': '', // ''
-        '5': '', // ''
-        '6': '', // ''
-        '7': '', // ''
-        '8': '', // ''
-        '9': '', // ''
-        '10': '', // ''
-        '11': '', // ''
-        '12': '', // ''
-        '13': '', // ''
-        '14': '', // ''
+        '1': 'Владелец', // 'Владелец'
+        '2': 'Исполнитель', // 'Исполнитель'
+        '3': 'Воронка', // 'Воронка'
+        '4': 'Этап', // 'Этап'
+        '5': 'Название', // 'Название'
+        '6': 'Телефоны', // 'Телефоны'
+        '7': 'Почты', // 'Почты'
+        '8': 'Клиент', // 'Клиент'
+        '9': 'ИНН клиента', // 'ИНН клиента'
+        '10': 'Примечание по лиду', // 'Примечание по лиду'
+        '11': 'Дата планируемой сделки', // 'Дата планируемой сделки'
+        '12': 'Планируемая сумма', // 'Планируемая сумма'
+        '13': 'Фактическая сумма', // 'Фактическая сумма'
+        '14': 'Файлы', // 'Файлы'
 };
 
 
 export const ALIAS = {
-        '1': '', // ''
-        '2': '', // ''
-        '3': '', // ''
-        '4': '', // ''
-        '5': '', // ''
-        '6': '', // ''
-        '7': '', // ''
-        '8': '', // ''
-        '9': '', // ''
-        '10': '', // ''
-        '11': '', // ''
-        '12': '', // ''
-        '13': '', // ''
-        '14': '', // ''
+        '1': 'owner', // 'Владелец'
+        '2': 'assignee', // 'Исполнитель'
+        '3': 'funnel', // 'Воронка'
+        '4': 'stage', // 'Этап'
+        '5': 'name', // 'Название'
+        '6': 'phones', // 'Телефоны'
+        '7': 'emails', // 'Почты'
+        '8': 'client', // 'Клиент'
+        '9': 'client_inn', // 'ИНН клиента'
+        '10': 'annotation', // 'Примечание по лиду'
+        '11': 'planned_date', // 'Дата планируемой сделки'
+        '12': 'planned_sum', // 'Планируемая сумма'
+        '13': 'actual_sum', // 'Фактическая сумма'
+        '14': 'files', // 'Файлы'
 };
 
 
diff --git a/typescript/app/application/sales-funnel/funnel/component/funnel-settings.component.ts b/typescript/app/application/sales-funnel/funnel/component/funnel-settings.component.ts
index 400c9e90c1..74a11af350 100644
--- a/typescript/app/application/sales-funnel/funnel/component/funnel-settings.component.ts
+++ b/typescript/app/application/sales-funnel/funnel/component/funnel-settings.component.ts
@@ -10,6 +10,8 @@ import TPropCategory = SalesFunnelFunnel.TPropCategory;
 import {TDealsRating} from "../module/deals-rating/deals-rating";
 import TLoadSettingsData = SalesFunnelFunnel.TLoadSettingsData;
 import {ELabelPosition} from "../../../../angular/component/amd-switch/amd-switch.component";
+import ArrayDataSource from "../../../../core/model/data-source/array.data-source";
+import AmdSelectFilter from "../../../../angular/component/amd-select/amd-select.filter";
 
 const TEMPLATE = `
 <amd-dialog data-wait-promise="$ctrl.waitPromise">
@@ -59,6 +61,13 @@ const TEMPLATE = `
                             >
                                 {{item.title}}
                             </amd-switch>
+                            <span ng-if="item.id == 3">
+                                <amd-multiselect
+                                    ng-if="$ctrl.fieldsSource"
+                                    ng-model="$ctrl.fileds"
+                                    data-source="$ctrl.fieldsSource"
+                                ></amd-multiselect>
+                            </span>
                         </div>
                     </amd-zend-control>
                 </div>
@@ -95,6 +104,10 @@ export class FunnelSettingsComponent implements IComponentController
 
     private eventActivitiesSettings: {is_active: boolean, title: string, id: number}[]
 
+    private fieldsSource: ArrayDataSource<{ text: unknown; value: string }>;
+
+    private fields: any[];
+
     constructor(
         @Inject(BackendService) private backendService: BackendService,
     ) {
@@ -120,6 +133,17 @@ export class FunnelSettingsComponent implements IComponentController
                         .map((item: any) => {
                             return {id: item.id, title: item.event_description, is_active: +item.is_active > 0}
                         });
+
+                    this.fields = data.fields;
+                    this.fieldsSource = new ArrayDataSource<{ text: unknown; value: string }>(
+                        new AmdSelectFilter(),
+                        Object.entries(this.fields).map(([key, value]) => {
+                            return {
+                                text: value.field_title,
+                                value: +key
+                            };
+                        })
+                    );
                 },
                 reason => (new MessageService()).alert(reason)
             );
@@ -133,7 +157,7 @@ export class FunnelSettingsComponent implements IComponentController
                 propCategories: this.propCategories,
                 dealsRating: this.dealsRating,
                 eventActivitiesSettings: this.eventActivitiesSettings.map(item => {
-                    return {id: item.id, is_active: item.is_active}
+                    return {id: item.id, is_active: item.is_active, additional: []}
                 })
             })
             .then(
diff --git a/typescript/app/application/sales-funnel/funnel/types.d.ts b/typescript/app/application/sales-funnel/funnel/types.d.ts
index 74b35ae627..9e1ead2831 100644
--- a/typescript/app/application/sales-funnel/funnel/types.d.ts
+++ b/typescript/app/application/sales-funnel/funnel/types.d.ts
@@ -44,6 +44,7 @@ export namespace SalesFunnelFunnel {
     };
 
     type TLoadSettingsData = {
+        fields: string[];
         eventActivities: any;
         settings: TSettings,
         propCategories: TPropCategory[],
