diff --git a/application/modules/analitics/controllers/ManagersBlocks/ListController.php b/application/modules/analitics/controllers/ManagersBlocks/ListController.php
index 90d85d90b5..0ed4b28ae7 100644
--- a/application/modules/analitics/controllers/ManagersBlocks/ListController.php
+++ b/application/modules/analitics/controllers/ManagersBlocks/ListController.php
@@ -1,122 +1,49 @@
 <?php
 
-use Analitics_Model_ManagersBlocks_List_Filter as Filter;
-use Analitics_Model_ManagersBlocks_List_FilterConfigurator as FilterConfigurator;
-use App_Filter_Excel_Converter as FilterConverter;
-use App_Filter_Controller_AbstractFilterController as FilterController;
-
-/**
- * Контроллер для работы со страницей "Брони заказов"
- * 
- * @category    Analitics
- * @package     Controller
- * @subpackage  ManagersBlocks
- * 
- * @author Anisimov A. <sanchezgeek@gmail.com>
- */
-class Analitics_ManagersBlocks_ListController extends FilterController
-{
-    /**
-     * Инициализация контроллера
-     */
-    public function init()
-    {
-        Analitics_Model_ManagersBlocks_Permissions::checkRules('refer');
-        
-        parent::init();
-    }
 
+use App\Controller\AngularJS\AbstractFilterPageController;
 
-    /**
-     * indexAction
-     */
-    public function indexAction()
-    {
-        Zend_Registry::set('statusString', 'Брони заказов');
+use App\Analytics\ManagersBlocks\Filter;
+use App\Analytics\ManagersBlocks\FilterConfigurator;
+use App\Development\TypeScript\FilterConfiguratorInterface;
 
-        $this->_getHeadLink()->appendStylesheet('/css/depot/depotdetailedmanager/transfer/form.css');
-        $this->_getHeadLink()->appendStylesheet('/css/analitics/managersblocks/index.css');
-        $this->_getHeadScript()->appendFile('/js/depot/product-measure-type-hint.js');
-        $this->_getHeadScript()->appendFile('/js/datatable/fieldshider.js');
-        $this->_getHeadScript()->appendFile('/js/analitics/managersblocks/filter/index.js');
-        $this->_getHeadScript()->appendFile('/js/analitics/managersblocks/controller/abstract.js');
-        $this->_getHeadScript()->appendFile('/js/analitics/managersblocks/controller/list.js');
-        $this->_getHeadScript()->appendFile('/js/form/form.js');
-        $this->_getHeadScript()->appendFile('/js/depot/depot_tooltip.js');
-
-        $this->view->controllerParams = array(
-            'routes' => array(
-                'setAnnotation' =>  $this->view->url(array('action' => 'set-annotation'), 'analyticsManagersBlocksList')
-            ),
-            'permissions' => Analitics_Model_ManagersBlocks_Permissions::getPermissionsArray(),
-            'filterConfig' => Analitics_Model_ManagersBlocks_List_Filter::getModelConfig()
-        );
-    }
 
+class Analitics_ManagersBlocks_ListController extends AbstractFilterPageController
+{
 
-    /**
-     * Возвращает конфигуратор фильтра
-     * 
-     * @return Manufacturers_Model_List_ReadyClaims_FilterConfigurator
-     */
-    protected function _getFilterConfigurator()
+    public function createFilter()
     {
-        if(!($this->_filterConfigurator instanceof FilterConfigurator)) {
-            $this->_filterConfigurator = new FilterConfigurator();
-        }
+        return new Filter();
 
-        return $this->_filterConfigurator;
     }
 
-
-    /**
-     * Возвращает фильтр контроллера
-     * В этом методе также данный фильтр можно конфигурировать
-     * 
-     * @return \App_Filter_AbstractFilter
-     */
-    protected function _createFilter()
+    public function getFilterConfigurator(): ?\App\Development\TypeScript\FilterConfiguratorInterface
     {
-        $filter = new Filter();
-        $this->_getFilterConfigurator()->setUpFilter($filter);
+        return new FilterConfigurator();
 
-        return $filter;
     }
 
-
-    /**
-     * Метод, вызываемые при создании excel-конвертера
-     * Вся конфигурация конвертера описывается здесь
-     * 
-     * @param App_Filter_AbstractFilter $filter Фильтр контроллера
-     * @param App_Filter_Excel_Converter $excelConverter Конвертер Filter->.xls
-     */
-    protected function _setUpFilterExcelConverter(\App_Filter_AbstractFilter $filter, FilterConverter $excelConverter)
+    public function indexAction()
     {
-        $this->_getFilterConfigurator()->setUpExcelMap($excelConverter);
-
-        // устанавливаем лимит на выгрузку
-        $excelConverter->getExcelWriter()->setRowsLimit(5000);
+        $this->view->getRequire()
+            ->setDataMain('/js/app/application/analytics/managers-block/list/index/config')
+            ->setParams('page.module', [
+                'dataTable' => [
+                    'urls' => $this->_getDatatableRoutes(),
+                ],
+                'access' => [
+                    'managersblocks' => [
+                        'refer' => App_Access::get('all', 'analitics>managersblocks>refer'),
+                    ]
+                ],
+            ]);
     }
 
-
-    /**
-     * Возвращает название excel-файла выгрузки
-     * 
-     * @return string
-     */
-    protected function _getExcelFileName()
+    public function init()
     {
-        return "Брони_заказов.xlsx";
+        parent::init();
+        Zend_Registry::set('statusString', 'Брони заказов');
+        \App\Access\Guard::assertEqual('analitics>managersblocks>refer', true);
     }
 
-
-    /**
-     * Запись примечания к depot_detailed
-     */
-    public function setAnnotationAction()
-    {
-        Analitics_Model_ManagersBlocks::getInstance()->setAnnotation($this->_getParam('row_id', false), $this->_getParam('annotation', ''));
-        exit;
-    }
 }
\ No newline at end of file
