Patch message: change

diff --git a/library/App/Logger/View/Clients/ViewGroup.php b/library/App/Logger/View/Clients/ViewGroup.php
index bbce8d8ca2..26e4447c83 100644
--- a/library/App/Logger/View/Clients/ViewGroup.php
+++ b/library/App/Logger/View/Clients/ViewGroup.php
@@ -95,14 +95,14 @@ class App_Logger_View_Clients_ViewGroup extends App_Logger_View_Abstract
                         <?php echo $codeTitle; ?>
                     </td>
                     <td class="loggerEventValue">
-                        <?php echo $oldValue; ?>
+                        <?= App_Db_Clients::obtain()->getClientFiledById("u_name", $oldValue); ?>
                     </td>
 <?php
                     if ($this->_eventType !== 1) {
 ?>
-                        <td class="loggerEventValue">
-                            <?php echo $newValue; ?>
-                        </td>
+                    <td class="loggerEventValue">
+                        <?= App_Db_Clients::obtain()->getClientFiledById("u_name", $newValue); ?>
+                    </td>
 <?php
                     }
 ?>
diff --git a/public/js/depot/depot_tooltip.js b/public/js/depot/depot_tooltip.js
index 3f00086cc3..5d75d8f910 100644
--- a/public/js/depot/depot_tooltip.js
+++ b/public/js/depot/depot_tooltip.js
@@ -127,6 +127,7 @@ var depot_tooltip = new (function() {
         $('body').prepend('<div id="depotTooltipContainer"></div>');
 
         $('body').on('click', '.depotTooltipWrapper', function(event){
+            console.log('here')
             // Клонируем текст текущего примечания (tooltip) из поля, для вставки в контейнер
             var tooltipTextInContainer = $(event.currentTarget).find('div#tooltipText').clone(true);
             // Есть ли в контейнере какие-либо tooltip-ы
@@ -157,8 +158,9 @@ var depot_tooltip = new (function() {
                 $('div#depotTooltipContainer').empty();
             }
 
-            $('body').on('click', 'div#depotTooltipContainer div#tooltipText[data-tooltip-id=' +tooltipTextInContainer.data('tooltip-id')+ ']',
+            $('body').on('click', 'body div#tooltipText[data-tooltip-id=' +tooltipTextInContainer.data('tooltip-id')+ ']',
                 function() {
+                console.log('here1')
                     this.remove();
                 }
             );
